### ISSUE TITLE
extractJson does not conform to JSON spec

### ISSUE THREAD

**Issue by obany** [iotaledger/iota.lib.js#231]
### Bug description
The extractJson method does not current conform to JSON spec https://tools.ietf.org/html/rfc7159.html

It's sanity checking near the beginning of the method looks for a "OD" opening brace and the loop checks for "QD" closing brace.

However, from the spec valid JSON objects can be any of the form:

```
object / array / false / true / null / number / string
```

The following are all valid JSON:

```javascript
JSON.parse("false")
JSON.parse("true");
JSON.parse("null");
JSON.parse("42.3");
JSON.parse("\"hello\"");
JSON.parse("[1, 2, 3]");
```

The current extractJson method precludes the use of these other root object types in its decoding.

Note that JSON.parse("undefined"); will throw an error.

### Steps To Reproduce
```javascript
iota.utils.extractJson([{
signatureMessageFragment: "JCVAQAWAQAXALC" 
}]
```

### Expected behaviour
method should return [1, 2, 3]

### Actual behavior
returns null

**Comment by chrisdukakis** [comment]
Hi @obany , nice suggestion! I think we can extend support to parse the valid forms you mentioned.

**Comment by chrisdukakis** [comment]
I've included support for the following cases in [`next`](https://github.com/iotaledger/iota.lib.js/tree/next/packages/extract-json):
- [x] `JSON.parse("false")`
- [x] `JSON.parse("true")`
- [x] `JSON.parse("null")`
- [x] `JSON.parse("\"hello\"")`
- [x] `JSON.parse("[1, 2, 3]")`
- [ ] `JSON.parse("42.3")`

Right now, it looks like only missing case is that of numbers. Not sure how to do it because of `9`s padding...

**Comment by obany** [comment]
Here is one potential solution for the number variant.

The limited precision available in javascript numbers means that a JSON number would have to fit in the first transaction of the trytes bundle.

In addition for it to be valid it must be entirely ASCII characters so pairs of trytes is a must.

To parse a valid json number with a regular RegEx we would use:

```
/^[-+]?[0-9]+(\.[0-9]+)?([eE][-+]?[0-9]+)?$/

From start of line: ^
Optionally match - or +: [-+]?
At least one digit: [0-9]+
Optionally match decimal places (...)?
      match dot \\.
      at least one digit: [0-9]+
Optionally match exponential (...)?
      match e or E: [eE]
      Optionally match - or +: [-+]?
      at least one digit: [0-9]+
To end of line: $
```

But this would not work in our case because we have trytes. 

So we just convert the regex to look for trytes instead of the ASCII characters and instead of looking for the end of line character look for 99 which must be the end of the number in the transaction.

```
/^(RA|PA)?(UA|VA|WA|XA|YA|ZA|9B|AB|BB|CB)+((SA)(UA|VA|WA|XA|YA|ZA|9B|AB|BB|CB)+)?((TC|OB)(RA|PA)?(UA|VA|WA|XA|YA|ZA|9B|AB|BB|CB)+)?99/

From start of line: ^
Optionally match - or +: (RA|PA)?
At least one digit: (UA|VA|WA|XA|YA|ZA|9B|AB|BB|CB)+
Optionally match decimal places (...)?
      match dot (SA)
      at least one digit: (UA|VA|WA|XA|YA|ZA|9B|AB|BB|CB)+
Optionally match exponential (...)?
      match e or E: (TC|OB)
      Optionally match - or +: (RA|PA)?
      at least one digit: (UA|VA|WA|XA|YA|ZA|9B|AB|BB|CB)+
To end of line: 99
```

With this RegEx we can modify the extractJson as follows:

```js
} else if (bundle[0].signatureMessageFragment.slice(0, 8) === 'BDID9D9D') {
	return 'null'
} else if (/^(RA|PA)?(UA|VA|WA|XA|YA|ZA|9B|AB|BB|CB)+((SA)(UA|VA|WA|XA|YA|ZA|9B|AB|BB|CB)+)?((TC|OB)(RA|PA)?(UA|VA|WA|XA|YA|ZA|9B|AB|BB|CB)+)?99/.test(bundle[0].signatureMessageFragment)) {
	return trytesToAscii(bundle[0].signatureMessageFragment.match(/^(.*)99/)[1]);
} else {
	throw new Error(errors.INVALID_JSON)
}
```

Since the more common cases like object, array are checked for earlier, this more expensive regex would not be called too often.

**Comment by chrisdukakis** [comment]
Fixed in #270
